### PR TITLE
Create API_design.md

### PR DESCRIPTION
**Pull request by KaoWeiCheng**
Please advise and guide me.
resolve https://github.com/No-57/feature/issues/3

### CODE REVIEW

File: API_design.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+## Header
+
+### login_header
+```
+Host: api.orange.com
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
+
```
Comment (shadow3x3x3) [inline API_design.md]: useless empty line

File: API_design.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+## Header
+
+### login_header
+```
+Host: api.orange.com
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
```
Comment (shadow3x3x3) [inline API_design.md]: Most of time, we don't specific the `User-Agent`.

File: API_design.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+```
+### guest_header
+`Allow anonymous access without requiring authentication headers`
```
Comment (shadow3x3x3) [inline API_design.md]: Do not use `` for normal sentense.

File: API_design.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+`https://api.orange.com/v1/products?limit=10&max_id=3`
+Retrieve 10 records starting from ID 3
+So, the frontend needs to update max_id with each request
+
+## Grid Table API Examples
+### URI
+`https://api.orange.com/v1/products`
+### Example 1（Get all data）：
+Request ：`Get  https://api.orange.com/v1/products`
```
Comment (shadow3x3x3) [inline API_design.md]:
```suggestion
Request: `GET https://api.orange.com/v1/products`
```

File: API_design.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+Response：
+```json
```
Comment (shadow3x3x3) [inline API_design.md:56]: Add an empty line between sentense and block.

File: API_design.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+        }      
+    ]
+}
+```
+
+### Errors example
+```json
+{
+    "code":400,
+    "errors": {
```
Comment (shadow3x3x3) [inline API_design.md:181]: If using error**s** key, the value of the key should be an Array.

File: API_design.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+    "code":200,
+    "data": [
+        {
+            "id": 2,
+            "name": "Macbook",
+            "price": "30000",
+            "description": "the newst M2 Macbook 8G/256G",
+            "like": "N",
+            "image_url": "XXXXX",
+            "createAt": "2023-10-07 12:00:00:000",
```
Comment (shadow3x3x3) [inline API_design.md]: Different naming style, please be consistency.

**Review by shadow3x3x3 — CHANGES_REQUESTED** [review]

File: API_design.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+max_id：Absolute position`
+
+`https://api.orange.com/v1/products?limit=10&max_id=3`
+Retrieve 10 records starting from ID 3
+So, the frontend needs to update max_id with each request
+
+## Grid Table API Examples
+### URI
+`https://api.orange.com/v1/products`
+### Example 1（Get all data）：
```
Comment (shadow3x3x3) [inline API_design.md]: I think `Example` is redundant, just use "Get all data" instead.

Even better, using "Get Products".

File: API_design.md [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+`Allow anonymous access without requiring authentication headers`
+```
+
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
+
+```
+
+### Methods：GET，POST
```
Comment (shadow3x3x3) [inline API_design.md]: Don't pollute with Chinese comma "，".

File: API_design.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
+ # Restful API
```
Comment (shadow3x3x3) [inline API_design.md]: A wired space here.

```suggestion
# Restful API
```

File: API_design.md [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+```
+
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
+
+```
+
+### Methods：GET，POST
+`https://api.orange.com/v1/products`
```
Comment (shadow3x3x3) [inline API_design.md]: The protocol and domain are not required for API Doc. The specific API URL is enough.

```suggestion
`/v1/products`
```

**Review by shadow3x3x3 — CHANGES_REQUESTED** [review]

**Comment by shadow3x3x3** [comment]
ping @Vince-Ku 

Do you have any comments on these API design? In particular, on iOS Dev.

**Comment by Vince-Ku** [comment]
> ping @Vince-Ku
> 
> Do you have any comments on these API design? In particular, on iOS Dev.

Currently, I only need two API formats about `Products` on the `Home Page`.
- GET /products ( pagination supported is required )
- GET /products?name=xx ( fuzzy search is required )

**Optional**

- GET /{user_id}/products?field={like, xx}
( we can still discuss where the favorite products originate from, Client or Server )

Except for those formats, others can wait.

**Comment by Vince-Ku** [comment]
Additionally, I wanna mention the concept of `Brand` and `Type`, both are also important in this project, cause I need to show all of them on the Home page.

ping @asdfg429, maybe we can discuss how to store `Brand` and `Type` in the database.

File: API_design.md (hunk 2 above)
Comment (KaoWeiCheng) [reply API_design.md]: Oh oh i got it

**Review by shadow3x3x3 — APPROVED** [review]

**Pull request merged**